src/Glpi/Asset/Capacity/HasImpactCapacity.php

```diff
@@ -0,0 +1,105 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset\Capacity;
+
+use CommonGLPI;
+use Config;
+use Impact;
+use ImpactItem;
+
+class HasImpactCapacity extends AbstractCapacity
+{
+    public function getLabel(): string
+    {
+        return Impact::getTypeName(1);
+    }
+
+    public function getIcon(): string
+    {
+        return Impact::getIcon();
+    }
+
+    public function isUsed(string $classname): bool
+    {
+        //NOTE: This doesn't consider cases where assets exist inside graphs on other asset types. Maybe there is no good solution since it involves scanning every graph.
+        return parent::isUsed($classname)
+            && countElementsInTable(ImpactItem::getTable(), ['itemtype' => $classname]) > 0;
+    }
+
+    public function getCapacityUsageDescription(string $classname): string
+    {
+        $impact_count = countElementsInTable(ImpactItem::getTable(), ['itemtype' => $classname]);
+        $asset_count = $this->countAssets($classname);
+
+        return sprintf(__('%1$s impact graphs attached to %2$s assets'), $impact_count, $asset_count);
+    }
+
+    public function onClassBootstrap(string $classname): void
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        $CFG_GLPI['impact_asset_types'][$classname] = $classname::getDefinition()->fields['picture'];
+
+        $enabled_types = json_decode(Config::getConfigurationValue('core', Impact::CONF_ENABLED), true) ?? [];
+        if (!in_array($classname, $enabled_types, true)) {
+            $enabled_types[] = $classname;
+            Config::setConfigurationValues('core', [Impact::CONF_ENABLED => json_encode($enabled_types)]);
+        }
+
+        CommonGLPI::registerStandardTab($classname, Impact::class, 2);
+    }
+
+    public function onCapacityDisabled(string $classname): void
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        unset($CFG_GLPI['impact_asset_types'][$classname]);
+
+        $enabled_types = json_decode(Config::getConfigurationValue('core', Impact::CONF_ENABLED), true) ?? [];
+        $enabled_types = array_diff($enabled_types, [$classname]);
+        Config::setConfigurationValues('core', [Impact::CONF_ENABLED => json_encode($enabled_types)]);
+
+        $impact_item = new ImpactItem();
+        $impact_item->deleteByCriteria([
+            'itemtype' => $classname,
+        ], true, false);
```

Comment:
It seems ImpactRelation values are not cleared when an item item is deleted.
The impact code is a bit old and probably doesn't do cascade deletion.

Reply:
Should be fixed in a different PR I think since it probably is a bug on 10.0.X